### ISSUE TITLE
Start subcommand should verify data folder wallet and mode match on startup

### ISSUE THREAD

**Issue by placer14** [OpenBazaar/openbazaar-go#954]
As a user of the server, I want the process to protect my data directory from becoming corrupted when I accidentally start the server with a different wallet than what the data directory was created with.

When the `start` command starts the server, it will default to a BTC wallet but a BCH or ZEC wallet may be used instead. When an initialized data directory is provided to the server on startup but uses a different wallet than what the directory was initialized with there is a high chance of the directory becoming corrupted.

This issue should deliver the following:

- [ ] initializing the repo directory should set a canonical indicator for which wallet was used so that future startup can verify this value
- [ ] the `start` command should consider this value and exit when it does not match the wallet being started by the server
- [ ] this issue should handle the case where the canonical indicator is not yet set in the repo directory and should discuss/agree on a way to infer the current configuration of the repo directory and set the canonical indicator appropriate

The indicator might be a value in `<data>/config` but should be human-readable.

**Comment by placer14** [comment]
Given I've seen some cases where running the BCH wallet against the repo folder that was originally started with a BTC wallet and vice-versa causes the repo to corrupt, might this be worth adding?

**Comment by BazaarDog** [comment]
If a data folder exists or is specified, couldn't the currency be taken from the data folder?

**Comment by placer14** [comment]
Yes, it would. I'm just specifying that the indicator should be protected and formal instead of something we infer from some data somewhere.

**Comment by cpacia** [comment]
I think the coming multiwallet support will probably make this a moot issue. Not sure if it's worth addressing for the short term.
